zerver/views/message_fetch.py

```diff
@@ -808,13 +824,37 @@ def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
             MAX_MESSAGES_PER_FETCH,
         ))
 
-    if user_profile.realm.email_address_visibility != Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+    # Web public queries are only allowed for limited types of narrows.
```

Comment:
TODO: Allow unauthenticated narrow for streams and topics, if the stream is web-public.

This will come in next commit.

Reply:
Sounds good.